dns_sd_avahi.c

```diff
@@ -60,6 +60,39 @@ void dnssd_free_discovery_data(struct dns_sd_discovery_data *d)
 	free(d);
 }
 
+static void __avahi_process_resolved(struct dns_sd_discovery_data *ddata, const AvahiAddress *addr,
+				     const char *host_name, const uint16_t port)
+{
+       /* Avahi is multi-threaded, so lock the list */
+       iio_mutex_lock(ddata->lock);
+       ddata->resolved++;
+
+       /* Find empty data to store things*/
+       while (ddata->next)
+               ddata = ddata->next;
+
+       /* link a new placeholder to the list */
+       avahi_address_snprint(ddata->addr_str,
+                       sizeof(ddata->addr_str), addr);
+       memcpy(ddata->address, addr, sizeof(*addr));
```

Comment:
This throws a warning in https://app.codacy.com/gh/analogdevicesinc/libiio/pullRequest?prid=6384953 since we no longer know the length of these...

Reply:
Will look at this. Honestly that code was just copy past

Reply:
Hmm, maybe I'm missing something oblivious, but I'm not getting this warning. `const AvahiAddress *addr` and `ddata->address` are pointers to the same data type so I'm not really seeing the issue here... I mean, I get the warning can be useful to double check this line (and potentially find a bug) but I'm not really seeing any issue with this particular line of code... Maybe if I change the pattern to `memcpy(ddata->address, addr, sizeof(*ddata->address)`, it shuts up the warning...

Reply:
So, this warning is still coming from codacy and I don't think there's much we can do about it (as the code stands). Alternatively, we can either just get rid of this line and the `address` variable (it looks is not really used) or make `address` a plain variable rather than a pointer...

Reply:
>So, this warning is still coming from codacy and I don't think there's much we can do about it 

I think it's that inside that function, it doesn't know the lengths. Since humans do, just do something like:

https://github.com/analogdevicesinc/libiio/blob/master/tests/iio_common.c#L66

Reply:
Ahh nice! I could have figured that we could somehow ignore the warning 🤦...